trace/obfuscate.go

```diff
@@ -1,40 +1,204 @@
 package trace
 
 import (
+	"reflect"
+	"regexp"
+	"strings"
+
 	pb "github.com/akitasoftware/akita-ir/go/api_spec"
 	"github.com/akitasoftware/akita-libs/spec_util"
 	. "github.com/akitasoftware/akita-libs/visitors"
 	vis "github.com/akitasoftware/akita-libs/visitors/http_rest"
+	"github.com/akitasoftware/go-utils/sets"
 	"github.com/postmanlabs/postman-insights-agent/printer"
 )
 
-func obfuscate(m *pb.Method) {
+type Obfuscator struct {
+	SensitiveDataKeys          sets.Set[string]
+	SensitiveDataValuePatterns sets.Set[*regexp.Regexp]
+}
+
+func NewObfuscator() *Obfuscator {
+	// Load obfuscation config using relative path to load the config file.
+	config, err := configFromFile("obfuscation_config.yaml")
+	if err != nil {
+		printer.Errorf("failed to load obfuscation config: %v\n", err)
+		return nil
+	}
+
+	sensitiveDataRegex := make([]*regexp.Regexp, len(config.SensitiveValueRegexes))
+
+	// Compile regex patterns.
+	for i, pattern := range config.SensitiveValueRegexes {
+		regex, err := regexp.Compile(pattern)
+		if err != nil {
+			printer.Errorf("failed to compile regex pattern %s: %v\n", pattern, err)
+			return nil
+		}
+		sensitiveDataRegex[i] = regex
+	}
+
+	return &Obfuscator{
+		SensitiveDataKeys:          sets.NewSet(config.SensitiveKeys...),
+		SensitiveDataValuePatterns: sets.NewSet(sensitiveDataRegex...),
+	}
+}
+
+func (o *Obfuscator) ObfuscateData(m *pb.Method) {
```

Comment:
I think the distinction between the names RedactData and ObfuscateData is not clear,

Can we say something like 'ZeroData' or 'ClearData' instead of 'Obfuscate'?  'Redact' is fine as-is; I just want to try avoiding synonyms that are too close in meaning.